### ISSUE TITLE
Direct Python Usage

### ISSUE THREAD

**Issue by mkorpela** [MarketSquare/robotframework-browser#45]
**Is your feature request related to a problem? Please describe.**
[PageObject](https://martinfowler.com/bliki/PageObject.html) and related [patterns](https://www.cypress.io/blog/2019/01/03/stop-using-page-objects-and-start-using-app-actions/), [issues](https://blog.getgauge.io/are-page-objects-anti-pattern-21b6e337880f) and [comments](https://johnfergusonsmart.com/page-objects-that-suck-less-tips-for-writing-more-maintainable-page-objects/) suggest that direct Python use will be a big part of this libraries success.

**Describe the solution you'd like**
I want a simple way to access and use library from python.

We need editor support for keyword completions.

**Describe alternatives you've considered**
Direct Robot Framework side use will severely limit library usability as most automation work will happen in python code. Although tests are written in Robot Framework.

**Additional context**
Hyvää Juhannusta

- [x] autocompletions @Snooz82 contributed (@xylix cherry picked) in ed26361d6c21ba368f6d9b3ad31b4aa8277e1854
- [x] Direct python usage without runtime dependencies on RF  #149 
  - [x] Added a detection for is robot running and a backup output path for library if robot isn't running 
    - Currenty requires utest/output to exist in directory where library is started and used
  - [x] Added robot logger mocking if robot is not running

**Comment by mkorpela** [comment]
I also think that pytest should be treated as a first class testing interface. We can support both: unit and acceptance test.

**Comment by yanne** [comment]
One problem for direct usage that I see is that the module name is "somewhat" generic.

Disregarding that, what does "simple" mean to you?
Would e.g. 

```
import Browser

chrome = Browser.open_browser()
chrome.go_to()
chrome.get_text()
```

be simple enough?

I guess the signatures would be the same as with RF, because e.g. with pytest you can just do the assertions yourself.

**Comment by aaltat** [comment]
The methods that are keywords are available in the library public API, with PythonLibCore. Meaning, this should work already now: 
```
import Browser

chrome = Browser.open_browser()
chrome.go_to()
chrome.get_text()
```
The problem is that IDE, example PyCharm, uses some static analysis and can not automatically complete the keywords.  The solution that I have been thinking in the [SL side](https://github.com/robotframework/SeleniumLibrary/pull/1591) is to build [python stub](https://www.python.org/dev/peps/pep-0484/#stub-files) files with [Mypy](https://mypy.readthedocs.io/en/stable/stubs.html).

**Comment by mkorpela** [comment]
Stub could work.. You could also be In princible a little bit hacky and not expose assertion engine syntax.

**Comment by Snooz82** [comment]
Hi,

I HACKED a little bit.
And it is really super HACKY.
The resulting `__init__.pyi` is useful.

We should really put all of our Enums into one module, so that all can be imported ad once...

Please don´t judge me!
i just wanted to have some completion and work on the SeleniumCompatibility Layer.
i know that this can be optimized a lot!!!

[generated __init__.pyi](https://github.com/Snooz82/robotframework-browser/blob/SeleniumLibraryMigration/Browser/__init__.pyi)

[super ugly script](https://github.com/Snooz82/robotframework-browser/blob/SeleniumLibraryMigration/gen_stub.py)

**Comment by xylix** [comment]
@Snooz82 Just letting you know, I merged the script in #149 along with some other improvements to raw python usage of the library.

**Comment by mkorpela** [comment]
This is in my opinion now done.
